test/micro/case/result_test.rb

```diff
@@ -169,6 +169,64 @@ def test_the_on_failure_hook
     assert_equal(2, counter)
   end
 
+  def test_the_on_unknown_hook
+    number = rand(1..1_000_000)
+
+    failure_result = failure_result(value: { number: number }, type: :not_mapped, use_case: Micro::Case.new({}))
+
+    failure_result
+      .on_failure(:a)
```

Comment:
@tomascco as Ruby is interpreted this line won't raise an exception, but, the `on_failure` implementation uses a `yield`, that is, the interpreter will raise an error (LocalJumpError) if the result type was `:b`. Follow a suggestion to fix this.

```suggestion
      .on_failure(:a) { raise }
```

Reply:
Oh, I think I got it now. Thanks :smile: